### PR TITLE
Support read only content

### LINKED ISSUE #11605: Support a read-only behavior in content management screen
### Is your feature request related to a problem? Please describe.
I have a need to give a specific role permission to edit some content types but view only other content types. To do that, I created a content type X and want to give U role access to ONLY view content (i.e., "View own X", "View X by others", "Preview X by others", and "List X content item(s) owner by all users"). These rules were not enough to show X in the manage contents screen. After testing, it turned out that in order for X to show up, I had to give "Edit X" permission which is not what I need here since I am trying to give read-only view.

Also, when I add X on AdminMenu, it does not show up unless the user has "publish or unpublished X" rule. At the very least for constancy, the same logic used in the manage admin screen should also apply on the AdminMenu. Sp if X appears on content-management it should also appear on AdminMenu and vice-versa.

### Describe the solution you'd like
I think both AdminMenu and content-management screens should be viewable if the user have "View own X", "View X by others", "Preview X by others", or "List X content item(s) owner by all users". Any of these permission should be enough to display content. View rules should be implied if one have edit, delete or publish permission "you can't delete, edit or publish if you can't view something". Making this change will easily add a view-only support to the contents in admin. I can't think of a good reason to not support view-only support in OC.

### Describe alternatives you've considered
Have not came up with a good iteam-

### PR DESCRIPTION
**Pull request by CrestApps**
Add support for read-only view and fix #11605

### CODE REVIEW

**Review by sebastienros — APPROVED** [review]
Can someone try it?

**Comment by hyzx86** [comment]
Wow, THAT's what I need,
I created a type that maps a CustomUserSettings type

**Comment by Skrypt** [comment]
I prefer merging this one after 1.4 unless someone tested this PR.

File: src/OrchardCore.Modules/OrchardCore.Contents/Views/Content.SummaryAdmin.cshtml [hunk 1]
```diff
@@ -61,14 +62,19 @@
             }
             @if (Model.ActionsMenu != null)
```
Comment (sebastienros) [inline src/OrchardCore.Modules/OrchardCore.Contents/Views/Content.SummaryAdmin.cshtml:63]: `&& Model.ActionsMenu.HasItems`

File: src/OrchardCore.Modules/OrchardCore.Contents/Views/Content.SummaryAdmin.cshtml [hunk 2]
```diff
@@ -82,3 +88,22 @@
...
+    public bool HasHtml(IHtmlContent content)
+    {
+        if (content is HtmlString htmlString)
+        {
+            return !String.IsNullOrWhiteSpace(htmlString.Value?.Trim());
+        }
+
+        using var writer = new System.IO.StringWriter();
+
+        content.WriteTo(writer, System.Text.Encodings.Web.HtmlEncoder.Default);
```
Comment (sebastienros) [inline src/OrchardCore.Modules/OrchardCore.Contents/Views/Content.SummaryAdmin.cshtml:104]: If you really need to render the menu:
- HtmlEncoder should be resolved by DI. (@inject HtmlEncoder HtmlEncoder)
- And get the stringbuilder from a pool, passed to the StringWriter.

File: src/OrchardCore.Themes/TheAdmin/Views/Message.cshtml [hunk 3]
```diff
@@ -1,4 +1,9 @@
 @{
+    if (Model.Type == null)
```
Comment (sebastienros) [inline src/OrchardCore.Themes/TheAdmin/Views/Message.cshtml:2]: Ideally log this error, this is unusual

File: src/OrchardCore.Modules/OrchardCore.Contents/Views/Content.SummaryAdmin.cshtml (hunk 1 above)
Comment (CrestApps) [reply src/OrchardCore.Modules/OrchardCore.Contents/Views/Content.SummaryAdmin.cshtml]: `Model.ActionsMenu.HasItems` did the trick and `HasHtml` is no longer needed! Great feedback

**Review by Skrypt — APPROVED** [review]

**Pull request merged**